### PR TITLE
CB-15764 Freeipa should implement ApplicationFlowInformation.handleFlowFail

### PR DESCRIPTION
**Pull request by sodre90**

### CODE REVIEW

File: freeipa/src/main/java/com/sequenceiq/freeipa/FreeIpaFlowInformation.java [hunk 1]
```diff
@@ -41,4 +52,14 @@ public List<Class<? extends FlowConfiguration<?>>> getTerminationFlow() {
         return TERMINATION_FLOWS;
     }
 
+    @Override
+    public void handleFlowFail(FlowLog flowLog) {
+        Stack stack = stackService.getStackById(flowLog.getResourceId());
+        LOGGER.info("Handling failed freeipa flow {} for {}", flowLog, stack);
```
Comment (lturcsanyi) [inline freeipa/src/main/java/com/sequenceiq/freeipa/FreeIpaFlowInformation.java:58]: stack -> stack.getId()?

**Review by lturcsanyi — APPROVED** [review]

File: freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/common/Status.java [hunk 2]
```diff
@@ -117,4 +138,37 @@ public boolean isVerticallyScalable() {
         return !FREEIPA_VERTICALLY_NON_SCALABLE_STATUSES.contains(this);
     }
 
+    //CHECKSTYLE:OFF: CyclomaticComplexity
```
Comment (lajosrodek) [inline freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/common/Status.java:141]: Sanity check: Is this needed? If so, there is no corresponding comment to re-enable the rule validation.

File: freeipa-api/src/test/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/common/StatusTest.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+                assertEquals(status, result);
+            }
+        }
+    }
+
+    @Test
+    public void inProgressTest() {
+        for (Status status : Status.values()) {
+            if (status.name().contains("REQUESTED") || status.name().contains("IN_PROGRESS")) {
+                if (!status.isInProgress()) {
```
Comment (lajosrodek) [inline freeipa-api/src/test/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/common/StatusTest.java:27]: Nit: Possibly collapse the two `if` statements into a single one.

**Review by lajosrodek — CHANGES_REQUESTED** [review]
Only some minor bits.

File: freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/common/Status.java [hunk 4]
```diff
@@ -117,4 +138,37 @@ public boolean isVerticallyScalable() {
...
+    public boolean isInProgress() {
```
[the remark this answers is not in the document]
Comment (sodre90) [reply freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/common/Status.java:142]: Thanks, I missed it!

**Review by lajosrodek — APPROVED** [review]
Thanks, LGTM

**Review by tiborpopovics — APPROVED** [review]

**Comment by akanto** [comment]
Isn't it related to: https://jira.cloudera.com/browse/CB-19969

**Pull request merged**